integration-tests/utils.py

```diff
@@ -351,16 +364,32 @@ def asyncify(f):
 
         # update bundle config
         data_path = os.path.join(bundle_dir, 'bundle.yaml')
-        with open(data_path, 'r') as f:
+        with open(data_path) as f:
             data = yaml.load(f)
-        if snap_channel:
-            for app in ['kubernetes-master', 'kubernetes-worker']:
-                options = data['services'][app].setdefault('options', {})
-                options['channel'] = snap_channel
-        data['services']['kubernetes-worker'].setdefault('options', {})['labels'] = 'mylabel=thebest'
-        yaml.Dumper.ignore_aliases = lambda *args: True
-        with open(data_path, 'w') as f:
-            yaml.dump(data, f, default_flow_style=False)
+            if os.getenv('http_proxy') and os.getenv('https_proxy'):
```

Comment:
This is a pretty big chunk of code to be copypasta'd.  Probably worth putting it in a function.  I mean, you made `arch` a function, but not all this?  ;)

Reply:
Easy!